### PR TITLE
Feat: add support for variable set assignment resource

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
Part of #829 

### Solution

1. Added the assignment resource.
2. Added an API call.
3. Added unit tests.
4. Added acceptance tests.
5. Added integration tests.
6. Updated examples.

### CODE REVIEW

**Comment by sagilaufer1992** [comment]
there are usages of both `variable` and `configuration` sets. can we make use of a single term? preferably the one used for single variables

File: env0/resource_variable_set_assignment.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	return &schema.Resource{
+		CreateContext: resourceVariableSetAssignmentCreate,
+		UpdateContext: resourceVariableSetAssignmentUpdate,
+		ReadContext:   resourceVariableSetAssignmentRead,
+		DeleteContext: resourceVariableSetAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"scope": {
+				Type:             schema.TypeString,
+				Description:      "the resource(scope) type to assign to. Valid values: 'template', 'environment', 'workflow', 'organization', 'project'",
```
Comment (sagilaufer1992) [inline env0/resource_variable_set_assignment.go:27]: the allowed values are: organization, project, template, module, environment, deployment
https://docs.env0.com/reference/configuration-find-set-assignments-by-scope see what the assignmentScope is allowed to be.

**Review by sagilaufer1992 — COMMENTED** [review]
looks great, left some comments

**Comment by TomerHeber** [comment]
> there are usages of both `variable` and `configuration` sets. can we make use of a single term? preferably the one used for single variables

yeah for the printing of messages. Sure thing.

**Comment by TomerHeber** [comment]
> looks great, left some comments

@sagilaufer1992 - thanks!
This is still work in progress. I will add the rest today.

File: env0/resource_variable_set_assignment.go (hunk 1 above)
Comment (TomerHeber) [reply env0/resource_variable_set_assignment.go:27]: thank you. will fix.

File: tests/integration/033_variable_set/main.tf [hunk 2]
```diff
@@ -70,3 +70,9 @@ resource "env0_variable_set" "project_scope" {
     format = "text"
   }
 }
+
+resource "env0_variable_set_assignment" "assignment" {
+  scope    = "project"
+  scope_id = env0_project.project.id
+  set_ids  = var.second_run ? [env0_variable_set.org_scope.id] : [env0_variable_set.project_scope.id]
```
Comment (sagilaufer1992) [inline tests/integration/033_variable_set/main.tf:77]: the test seems a bit missing... can we check the outcome of that? using output or some other method, to see that the assignment doesn't exist on the org after the second run, and exists where it should after on each run?

**Review by sagilaufer1992 — APPROVED** [review]
LGTM
requested to update the integration test to make sure the expected outcome is achieved, not just alter the resource

File: tests/integration/033_variable_set/main.tf (hunk 2 above)
Comment (TomerHeber) [reply tests/integration/033_variable_set/main.tf:77]: This is tested in the acceptance tests.
The integration tests are limited in their functionality and are more of a sanity just to make sure things really work...
I don't think there's an easy way to test what you're asking with the current integration tests... /:

Comment (sagilaufer1992) [reply tests/integration/033_variable_set/main.tf:77]: Alright. As long as there is a test for it

**Pull request merged**